086-monitoring-of-custom-resources.md

```diff
@@ -0,0 +1,61 @@
+# Monitoring of custom resources
+
+In https://github.com/strimzi/strimzi-kafka-operator/issues/9802 there was a discussion about how to monitor the state of a custom resources (CR) like `KafkaTopic`, `KafkaUser`, ... .
+For a simple and Kubernetes native monitoring, [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) (KSM) can be used which utilizes e.g. the status object of a Kubernetes resource.
+
+## Current situation
+
+Strimzi provides metrics for several use cases but currently not for the state of `KafkaTopic` and `KafkaUser` CRs.
+TODO need input for the status of other CRs and their metrics, e.g. `Kafka`, `StrimziPodSet`, `KafkaRebalance`...
+
+## Motivation
+
+Strimzi already provides example configurations being compatible with the prometheus-operator via e.g. `PodMonitor` for scraping the metrics and e.g. `PrometheusRules` for alerting.
+This is implemented by metrics-exporter via JMX or kafka-exporter.
+
+But in general, if a deployed Kubernetes resource is having a problem or a deprecated configuration is used, you want to get notified about that and can fix the problem, afterwards.
+Other tools like Flux handles the monitoring of CRs via KSM like documented in their [documentation](https://fluxcd.io/flux/monitoring/custom-metrics/).
+In addition Flux provides an [example repository](https://github.com/fluxcd/flux2-monitoring-example) for configuring KSM as part of [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) Helm chart.
+But the required configuration is just being part of the ConfigMap used by KSM as shown by @im-konge in this [issue](https://github.com/strimzi/strimzi-kafka-operator/issues/10276#issuecomment-2276088493).
+
+## Proposal
+
+So in order to have a monitoring in place for all CRs and implementing them in a scalable and Kubernetes native way KSM can be used.
+KSM is (mostly) deployed via a simple Kubernetes `Deployment` and querries the Kubernetes api for e.g. the status object of a Kubernetes resource indicating that the CR having a problem or contains a deprecated configuration.
+This can also be done for CRs exclusivly by using the `--custom-resource-state-config` flag which is helpful for e.g. `KafkaTopic`, `KafkaUser`, ... CRs.
+
+### Deployment
+
+Strimzi should provide guidance for existing fields in status object in order to indicate if a CR is having a problem or uses a deprecated configuration in the official documentation.
+By this users can extend monitoring of CRs on their own requirements.
+
+In general, an example ConfigMap should be provided in the examples directory of [strimzi/strimzi-kafka-operator](https://github.com/strimzi/strimzi-kafka-operator/tree/main/examples) specifying a good starting configuration.
```

Comment:
Should we be more specific here? I guess it will be somewhere as `https://github.com/strimzi/strimzi-kafka-operator/tree/main/examples/metrics/kube-state-metrics` probably?

Maybe you could include a simple sample of what the config map will look like to give a better idea? E.g. just for a single resource.